src/core/support/string.c

```diff
@@ -174,6 +188,27 @@ char *gpr_strjoin(const char **strs, size_t nstrs, size_t *final_length) {
   return out;
 }
 
+static void do_nothing(void *ignored) {}
+gpr_slice_buffer *gpr_strsplit(const char *str, const char *sep) {
```

Comment:
what about:

```c
void gpr_strsplit(const char *str, const char *sep, gpr_slice_buffer *dst);
```

or maybe one better:
```c
void gpr_slice_split(gpr_slice str, gpr_slice sep, gpr_slice_buffer *dst);
```

... the latter could do this without any allocations whatsoever, and given that metadata comes in the form of slices...